File: mobile_app/lib/domain/mode_two.dart
import 'package:rvi_analyzer/domain/default_configuration.dart';
import 'package:rvi_analyzer/domain/session_result.dart';

class ModeTwo {
  final String createdBy;
  final DefaultConfiguration defaultConfigurations;
  final SessionConfigurationModeTwo sessionConfigurationModeTwo;
  final List<SessionResult> results;
  final String status;

  ModeTwo(
      {required this.createdBy,
      required this.defaultConfigurations,
      required this.sessionConfigurationModeTwo,
      required this.results,
      required this.status});

  Map toJson() => {
        'createdBy': createdBy,
        'defaultConfigurations': defaultConfigurations.toJson(),
        'sessionConfigurationModeTwo': sessionConfigurationModeTwo.toJson(),
        'results': results,
        'status': status,
      };
}

class SessionConfigurationModeTwo {
  final String current;
  final String maxVoltage;
  final String passMinVoltage;
  final String passMaxVoltage;

  SessionConfigurationModeTwo(
      {required this.current,
      required this.maxVoltage,
      required this.passMinVoltage,
      required this.passMaxVoltage});

  Map toJson() => {
        'current': current,
        'maxVoltage': maxVoltage,
        'passMinVoltage': passMinVoltage,
        'passMaxVoltage': passMaxVoltage
      };
}

File: rvi-analyzer-server/src/main/java/com/rvi/analyzer/rvianalyzerserver/entiy/ModeTwo.java
package com.rvi.analyzer.rvianalyzerserver.entiy;

import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "ModeTwo")
public class ModeTwo {
    @Id
    private Long _id;
    @Column
    private String createdBy;
    @Column
    private DefaultConfiguration defaultConfigurations;
    @Column
    private SessionConfigurationModeTwo sessionConfigurationModeTwo;
    @Column
    private List<SessionResult> results;
    @Column
    private String status;
    @Column
    @CreatedDate
    private LocalDateTime createdDateTime;
    @Column
    @LastModifiedDate
    private LocalDateTime lastUpdatedDateTime;
}
